### ISSUE TITLE
lstm 

### ISSUE THREAD

**Issue by vinayakumarr** [tflearn/tflearn#52]
I have dataset (a part of dataset is given below)

3	3	373	27	9	615	9	16	10	34	0	8	0	199	65917	1319	122	402	319	183
3	3	77	12	4	66	4	3	0	5	0	14	3	50	106	139	38	164	53	109
3	3	86	6	2	6	2	0	0	1	0	25	0	4	284	77888	19	66	11	25
3	3	469	21	7	291	7	43	15	82	0	207	0	181	115646	59073	294	928	112	675
3	3	2090	21	7	4035	7	17	8	40	0	317	10	717	1033	25661	142	2054	1795	1023
3	3	691	18	6	597	6	30	16	61	0	245	18	273	719	2352305	213	1106	324	719
6	6	229	0	8	526	0	11	1	13	0	6	5	101	7246	2082	120	141	288	1570
3	3	1158	9	3	649	3	16	6	17	1	247	38	477	592	987626	82	1305	653	707
4	4	211	0	10	429	0	16	9	20	0	3	0	106	42725	27302	4280	133	477	1567

The first column is the target which has 9 classes and around 1803 features i am having it. 
I am trying to apply lstm for this could you please a provide a sample code for this. also i am trying to implement lstm in 

Two memory blocks with two cells each,
four memory blocks with two cells each,
four memory blocks with four cells each, and
eight memory blocks with four cells each.

**Comment by aymericdamien** [comment]
You need to format your data as follow:
```python
trainX = [
   [373 27 9 615 9 16 10 34 0 8 0 199 65917 1319 122 402 319 183],
   [77 12 4 66 4 3 0 5 0 14 3 50 106 139 38 164 53 109],
   ...
]
trainY = [3, 3, 3 ...]
trainY = tflearn.data_utils.to_categorical(Y1) # Change labels to one-hot vector
```
For a model, seems you can use the one in the [lstm example](https://github.com/tflearn/tflearn/blob/master/examples/nlp/lstm.py) (If these integers represent keys?) otherwise, if they are features, I suggest you normalize them by dividing all your values by the max one.

In case you have multiple labels, you can do two regressions:
```
...
net = tflearn.fully_connected(net, 2, activation='softmax')
regression1 = tflearn.regression(net, optimizer='adam',
                                                    loss='categorical_crossentropy')
regression2 = tflearn.regression(net, optimizer='adam',
                                                    loss='categorical_crossentropy')
net = tflearn.merge_outputs(regression1, regression2)

model = tflearn.DNN(net, clip_gradients=0., tensorboard_verbose=0)
model.fit(trainX, [trainY1, trainY2], show_metric=True, batch_size=128)
```
Result will be a merged layer of your prediction with regression1 and regression2. This can be changed according to your need (see merge layer), you could also take the average for example).

**Comment by vinayakumarr** [comment]
hi, According to you. I tried to implement lstm example (imdb example). The program is given below. It is generaing tensor shapes are not same. Please have a look for detailed error.

from __future__ import print_function
import numpy as np
from sklearn.cross_validation import train_test_split
import tflearn
import pandas as pd
from tflearn.data_utils import to_categorical, pad_sequences

print("Loading")
data = pd.read_csv('LargeTrain.csv')

X = data.iloc[:,1:1805]
y = data.iloc[:,0]


X_train, X_test, y_train, y_test = train_test_split(X, y,test_size=0.2,random_state=42)


print(X_train.shape)
print(X_test.shape)
print(y_train.shape)
print(y_test.shape)

print("Preprocessing")
X_train1 = X_train.values.T.tolist()
X_test1 = X_test.values.tolist()
y_train1 = y_train.values.T.tolist()
y_test1 = y_test.values.tolist()

# Data preprocessing
# Sequence padding
trainX = pad_sequences(X_train1, maxlen=200, value=0.)
testX = pad_sequences(X_test1, maxlen=200, value=0.)

# Converting labels to binary vectors
trainY = to_categorical(y_train, nb_classes=0)
testY = to_categorical(y_test, nb_classes=0)

# Network building
net = tflearn.input_data([None, 200])
net = tflearn.embedding(net, input_dim=20000, output_dim=128)
net = tflearn.lstm(net, 128)
net = tflearn.dropout(net, 0.5)
net = tflearn.fully_connected(net, 2, activation='softmax')
net = tflearn.regression(net, optimizer='adam',loss='categorical_crossentropy')

# Training
model = tflearn.DNN(net, clip_gradients=0., tensorboard_verbose=0)
model.fit(trainX, trainY, validation_set=(testX, testY), show_metric=True, batch_size=128)


![untitled](https://cloud.githubusercontent.com/assets/18217467/14934631/acb73bd6-0ed4-11e6-9433-b1c2f35486fa.png)
